kernel/core/net/net_wire.c

```
@@ -0,0 +1,212 @@
+
+    if (!frame || len < FL_NET_ETH_FRAME_HDR_LEN + FL_NET_IPV4_HDR_LEN_MIN || !ip_off ||
+        !ip_len || !dst_addr)
+        return 0;
+
+    if (!fl_net_wire_ethertype_is_ipv4(frame, len))
+        return 0;
+
+    off = (size_t)FL_NET_ETH_FRAME_HDR_LEN;
+    ip = frame + off;
+    iplen = (size_t)(((uint16_t)ip[2] << 8) | (uint16_t)ip[3]);
+    if (iplen < FL_NET_IPV4_HDR_LEN_MIN || off + iplen > len)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)